staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -472,16 +472,21 @@ func (w *watchCache) WaitUntilFreshAndList(ctx context.Context, resourceVersion
 		return nil, 0, "", err
 	}
 
-	// This isn't the place where we do "final filtering" - only some "prefiltering" is happening here. So the only
-	// requirement here is to NOT miss anything that should be returned. We can return as many non-matching items as we
-	// want - they will be filtered out later. The fact that we return less things is only further performance improvement.
-	// TODO: if multiple indexes match, return the one with the fewest items, so as to do as much filtering as possible.
-	for _, matchValue := range matchValues {
-		if result, err := w.store.ByIndex(matchValue.IndexName, matchValue.Value); err == nil {
-			return result, w.resourceVersion, matchValue.IndexName, nil
+	result, rv, index, err := func() ([]interface{}, uint64, string, error) {
```

Comment:
What did we gain from this closure?

Reply:
I'm also a bit confused by this change. Also, we should be able to return directly from the function call. EDIT: I guess you aren't because you intend to sort the results.

Reply:
Yeah - I did it to be able to put `sort` in just one place instead of two. Should get more clear once I finish this PR.